docs/chainlink-keepers/compatible-contracts.md

```diff
@@ -2,43 +2,78 @@
 layout: nodes.liquid
 section: ethereum
 date: Last Modified
-title: 'Making Keeper-compatible Contracts'
+title: 'Making Keepers-compatible Contracts'
 whatsnext:
   {
-    'Register for Upkeep': '/docs/chainlink-keepers/register-upkeep/',
+    'Register your Upkeep': '/docs/chainlink-keepers/register-upkeep/',
   }
 ---
 
 ## Overview
 
-This guide explains how to make smart contracts **Keeper-compatible**. You will learn about the `KeeperCompatibleInterface` and its functions with an example contract. The guide showcases the convenience that Keepers provide to developers. To take full advantage of the Keepers automation infrastructure, read all of the documentation to understand the features of Chainlink Keepers.
+Learn how to make smart contracts **Keepers-compatible** with the `KeeperCompatibleInterface` and its functions.
 
 **Table of Contents**
-
++ [Example Contract](#example-contract)
 + [Functions](#functions)
   + [`checkUpkeep` function](#checkupkeep-function)
     + [`checkData`](#checkdata)
     + [`performData`](#performdata)
   + [`performUpkeep` function](#performupkeep-function)
     + [`performData`](#performdata-1)
-+ [Example Contract](#example-contract)
++ [Best practices](#best-practices)
+  + [Revalidate `performUpkeep`](#revalidate-performupkeep)
+  + [Test your contract](#test-your-contract)
+
+
+## Example Contract
+
+Keepers-compatible contracts must meet the following requirements:
+
+* Import `KeepersCompatible.sol`. You can refer to the [Chainlink Contracts](https://github.com/smartcontractkit/chainlink/tree/develop/contracts/src) on GitHub to find the latest version.
+* Use the `KeepersCompatibleInterface` from the library to ensure your `checkUpkeep` and `performUpkeep`function definitions match the definitions expected by the Keepers Network.
+* Include a `checkUpkeep` function that contains the logic that will be executed off-chain to see if `performUpkeep` should be executed. `checkUpkeep` can use on-chain data and a specified `checkData` parameter to perform complex calculations off-chain and then send the result to `performUpkeep` as `performData`.
+* Include a `performUpkeep` function that will be executed on-chain when `checkUpkeep` returns `true`. Since `performUpkeep` is external, users are advised to revalidate conditions and performData.
```

Comment:
```suggestion
* Include a `performUpkeep` function that will be executed on-chain when `checkUpkeep` returns `true`. Because `performUpkeep` is external, users are advised to revalidate conditions and performData.
```
Use "since" to refer to time passed. Use "because" to refer to reason.